### PR TITLE
changes made to publish to maven

### PR DESCRIPTION
**Pull request by nswamy**
## Description ##
Changes that were made to publish the packages to Maven, some of the changes were already committed by the Maven Release plugin on my behalf - I have disabled this functionality. Currently it will only create a tag on the local repo when you run release-prepare.

In brief.
- We will build the MXNet backend during the compilation of init package only for `apache-release` profile.
- Disabled automatically committing to Git and instead only locally commit.
- Reduce the number of packages to be published to 3 platform specific platforms.
- Flatten pom files during the build phase -- otherwise the package cannot be used and we need to publish POM files.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scala-package/dev/compile-mxnet-backend.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+chkret pushd $MXNETDIR
+chkret git submodule update --init --recursive
+
+# don't want to overwrite an existing config file
+cp make/config.mk ./config.mk
+
+if [[ $PLATFORM == "osx-x86_64-cpu" ]];
+then
+    echo "Building MXNet Backend on MAC OS"
+    echo "ADD_CFLAGS += -I/usr/local/opt/opencv@2/include" >> ./config.mk
```
Comment (lanking520) [inline scala-package/dev/compile-mxnet-backend.sh]: Need to fix the opencv@2 here. Currently MXNet Python package use different openCV version

File: scala-package/pom.xml [hunk 2]
```diff
@@ -30,12 +30,13 @@
...
     <developerConnection>scm:git:git@github.com:apache/incubator-mxnet.git</developerConnection>
     <url>https://github.com/apache/incubator-mxnet</url>
-    <tag>mxnet-parent_2.11-1.2.0</tag>
+    <tag>HEAD</tag>
   </scm>
 
   <properties>
     <scala.version>2.11.8</scala.version>
     <scala.binary.version>2.11</scala.binary.version>
+    <build.platform />
```
Comment (lanking520) [inline scala-package/pom.xml]: Why we have it here, it seemed empty?

Comment (nswamy) [reply scala-package/pom.xml]: this is passed from the Makefile. ie., to say no default value

File: scala-package/dev/compile-mxnet-backend.sh (hunk 1 above)
Comment (nswamy) [reply scala-package/dev/compile-mxnet-backend.sh]: Yes, lets fix this in the next iteration during release. I don't know at the moment what will work.

Comment (nswamy) [reply scala-package/dev/compile-mxnet-backend.sh]: Here is the issue to track https://github.com/apache/incubator-mxnet/issues/1119

File: scala-package/dev/compile-mxnet-backend.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    echo "ADD_CFLAGS += -I/usr/local/cuda/include" >> ./config.mk
+    echo "ADD_LDFLAGS += -L/usr/local/cuda/lib64/  " >> ./config.mk
+    #update th nccl version approriately
+    echo "ADD_LDFLAGS += -L/lib/nccl/cuda-9.0/lib  " >> ./config.mk
+    eval "export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/lib/nccl/cuda-9.0/lib"
+    eval "export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH"
+    make -j$(cat /proc/cpuinfo | awk '/^processor/{print $3}' | tail -1)
+    echo "Building MXNet Backend on Linux GPU"
+else
+    echo "MY ALIEN OVERLOADS HAVE NOT TOLD WHAT TO DO FOR INVALID INPUT !!!"
```
Comment (marcoabreu) [inline scala-package/dev/compile-mxnet-backend.sh]: <3

**Comment by lanking520** [comment]
Thanks for your input, let's keep the issue tracked and solve that. @nswamy  LGTM + 1

File: scala-package/pom.xml [hunk 4]
```diff
@@ -30,12 +30,13 @@
     <connection>scm:git:git@github.com:apache/incubator-mxnet.git</connection>
     <developerConnection>scm:git:git@github.com:apache/incubator-mxnet.git</developerConnection>
     <url>https://github.com/apache/incubator-mxnet</url>
-    <tag>mxnet-parent_2.11-1.2.0</tag>
+    <tag>HEAD</tag>
```
Comment (marcoabreu) [inline scala-package/pom.xml]: Is this intended to point to the HEAD (master)?

Comment (nswamy) [reply scala-package/pom.xml]: Yes, It has to create a new tag every time.

Comment (marcoabreu) [reply scala-package/pom.xml]: I'm not very familiar with that setup, so please bare with me. What exactly is this supposed to do? I'd expect it to point to a certain branch, but this one would point to the master. What's the use-case here?

Comment (nswamy) [reply scala-package/pom.xml]: Its just telling what tagName should be used when releasing. when HEAD it uses the name of the artifact defined in the pom file + version. We are not using this anyway, like I said on the email I have disabled tagging on the remote repo

**Comment by lanking520** [comment]
@marcoabreu this one https://github.com/apache/incubator-mxnet/issues/11193

File: scala-package/dev/compile-mxnet-backend.sh (hunk 1 above)
Comment (marcoabreu) [reply scala-package/dev/compile-mxnet-backend.sh]: During release?

**Pull request merged**